pages/admin/students.js

```diff
@@ -0,0 +1,236 @@
+      await mutate(
+        [
+          ...data,
+          {
+            email: reformatEmail(newStudentEmail),
+            section: newStudentSection,
+            fname: newStudentFname,
+            lname: newStudentLname,
+            permNum: newStudentPermNum,
+          },
+        ],
+        false
+      );
+      await fetch("/api/students", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email: reformatEmail(newStudentEmail),
+          section: newStudentSection,
+          fname: newStudentFname,
+          lname: newStudentLname,
+          permNum: newStudentPermNum,
+        }),
+      });
+      await mutate();
+    },
+    [
+      newStudentEmail,
+      newStudentSection,
+      newStudentFname,
+      newStudentLname,
+      newStudentPermNum,
+    ]
+  );
+
+  const deleteStudent = useCallback(async (studentId) => {
+    console.log(studentId);
+    showToast(
+      `Deleted student ${data.find((u) => u._id === studentId)?.email}`
+    );
+    await mutate(
+      data.filter((u) => u._id !== studentId),
+      false
+    );
+    await fetch(`/api/students/${studentId}`, { method: "DELETE" });
+    await mutate();
+  }, []);
+
+  const columns = getColumnsWithActions((_, row) => {
+    return (
+      <Button variant="danger" onClick={() => deleteStudent(row._id)}>
+        Delete
+      </Button>
+    );
+  });
+  const handleFiles = (files) => {
+    var reader = new FileReader();
+    reader.onload = (e) => {
+      //Split csv file data by new line so that we can skip first row which is header
+      let jsonData = reader.result.split("\n");
+      jsonData.forEach(async (element, index) => {
+        if (index) {
+          //Split csv file data by comma so that we will have column data
+          const elementRaw = element.split(",");
+          if (element) {
+            let param = {
+              permNum: elementRaw[1],
+              email: reformatEmail(elementRaw[10]),
+              lname: elementRaw[4],
+              fname: elementRaw[5],
+              section: elementRaw[8],
+            };
+            await mutate(
+              [
+                ...data,
+                {
+                  permNum: param.permNum,
+                  email: param.email,
+                  lName: param.lName,
+                  fName: param.fName,
+                  section: param.section,
+                },
+              ],
+              false
+            );
+            await fetch("/api/students", {
+              method: "POST",
+              headers: {
+                "Content-Type": "application/json",
+              },
+              body: JSON.stringify({
+                permNum: param.permNum,
+                email: param.email,
+                lname: param.lname,
+                fname: param.fname,
+                section: param.section,
+              }),
+            });
```

Comment:
I wonder if it would be better to refactor this flow from the current:
```
for each row in json
	make a student object
	create student in database
```

to

```
students = json.mapFromRowToStudents()
for each student
	create student in database
```

I think it would certainly be easier to read/refactor/reuse the bottom flow than the top flow.

Reply:
@Ulramok that's a question for you

Reply:
I agree, I will update to use the bottom flow

Reply:
Okay I refactored the code to make use of the bottom flow, lmk if anything stands out